src/insert/spacefinder/spacefinder.ts

```diff
@@ -306,7 +306,10 @@ const newTestCandidate = (
 	candidate: SpacefinderItem,
 	opponent: SpacefinderItem,
 ): boolean => {
-	if (candidate.element === opponent.element) {
+	if (
+		candidate.element === opponent.element ||
+		opponent.element.contains(candidate.element)
```

Comment:
Would you mind explaining this please?

Reply:
This does looks weird, took a moment to remember, will add a comment.

So for example with key takeaways, the key takeaways block itself will be an opponent but will have descendants that are candidates. The descendant will be trying to avoid it's ancestor.

There's never a situation where want a candidate to avoid an ancestor element,   so we bypass the check.

Reply:
Thanks for updating! Much clearer